packages/go_router_builder/example/lib/go_relative.dart

```diff
@@ -0,0 +1,156 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: public_member_api_docs, unreachable_from_main
+
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+part 'go_relative.g.dart';
+
+void main() => runApp(const MyApp());
+
+/// The main app.
+class MyApp extends StatelessWidget {
+  /// Constructs a [MyApp]
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp.router(
+      routerConfig: _router,
+    );
+  }
+}
+
+/// The route configuration.
+final GoRouter _router = GoRouter(
+  routes: $appRoutes,
+);
+const TypedRelativeGoRoute<DetailsRoute> detailRoute =
```

Comment:
What is different between using this vs a regular TypedGoRoute?

Reply:
The example I gave isn't clearly explaining the purpose. Let me update this

Reply:
updated

Reply:
I still can't see the difference. Is this a way to reuse the typedGoRoute in multiple places in the routing tree?

Reply:
Yes. Without this you'd have to define routes like
SettingsFromHomeRoute
SettingsFromDashboardRoute

And those routes all build (or build page) the same screen

Reply:
yeah this makes sense. I will take another look